lib/dal/src/management/mod.rs

```diff
@@ -971,19 +991,30 @@ impl<'a> ManagementOperator<'a> {
         Ok(pending_operations)
     }
 
-    fn get_real_component_id(&self, placeholder: &String) -> ManagementResult<ComponentId> {
-        // Created component placeholders take priority over managed component
-        // placeholders (if you create a component with the same name as a
-        // managed component, references in updates, connections and actions
-        // will be to the created component, not the pre-existing managed
-        // component)
+    fn get_managed_component_id_opt(&self, placeholder: &str) -> Option<ComponentId> {
         self.component_id_placeholders
             .get(placeholder)
             .or_else(|| self.managed_component_id_placeholders.get(placeholder))
             .copied()
-            .ok_or(ManagementError::ComponentWithPlaceholderNotFound(
-                placeholder.to_owned(),
-            ))
+    }
+
+    fn get_managed_component_id(&self, placeholder: &str) -> ManagementResult<ComponentId> {
+        self.get_managed_component_id_opt(placeholder).ok_or(
+            ManagementError::ComponentWithPlaceholderNotFound(placeholder.to_owned()),
+        )
+    }
+
+    fn get_managed_or_external_component_id(
```

Comment:
This is how we support component ID in socket refs. We could expand this support to all places we look up components (i.e. just modify the existing get_real_component_id), but I wanted to be intentional about that--for example, I'm not sure whether we would want `ops.update` to be able to update arbitrary components in the graph.